File: C++/Binary Tree/inorder_traversal.cpp
// including libraries 
#include <bits/stdc++.h>

using namespace std;

// node 
struct node {
  int data;
  struct node * left, * right;
};

// vector for inorder traversal 
vector < int > inOrderTrav(node * curr) {
  vector < int > InOrder;
  stack < node * > s;
  while (true) {
    if (curr != NULL) {
      s.push(curr);
      curr = curr -> left;
    } else {
      if (s.empty()) break;
      curr = s.top();
      InOrder.push_back(curr -> data);
      s.pop();
      curr = curr -> right;
    }
  }
  return InOrder;
}

struct node * newNode(int data) {
  struct node * node = (struct node * ) malloc(sizeof(struct node));
  node -> data = data;
  node -> left = NULL;
  node -> right = NULL;

  return (node);
}

// driver code 
int main() {

  struct node * root = newNode(1);
  root -> left = newNode(2);
  root -> right = newNode(3);
  root -> left -> left = newNode(4);
  root -> left -> right = newNode(5);
  root -> left -> right -> left = newNode(8);
  root -> right -> left = newNode(6);
  root -> right -> right = newNode(7);
  root -> right -> right -> left = newNode(9);
  root -> right -> right -> right = newNode(10);

  vector < int > InOrder;
  InOrder = inOrderTrav(root);

  cout << "The InOrder Traversal is : ";
  for (int i = 0; i < InOrder.size(); i++) {
    cout << InOrder[i] << " ";
  }
  return 0;
}

File: Java/Binary Tree/inorder_traversal.java
// Java program for inorder traversals

import java.util.*;

// Structure of a Binary Tree Node
class Node {
	int data;
	Node left, right;

	Node(int v)
	{
		data = v;
		left = right = null;
	}
}

// Main class
class GFG {

	// Function to print inorder traversal
	public static void printInorder(Node node)
	{
		if (node == null)
			return;

		// First recur on left subtree
		printInorder(node.left);

		// Now deal with the node
		System.out.print(node.data + " ");

		// Then recur on right subtree
		printInorder(node.right);
	}

	// Driver code
	public static void main(String[] args)
	{
		Node root = new Node(1);
		root.left = new Node(2);
		root.right = new Node(3);
		root.left.left = new Node(4);
		root.left.right = new Node(5);
		root.right.right = new Node(6);

		// Function call
		System.out.println(
			"Inorder traversal of binary tree is: ");
		printInorder(root);
	}
}